infra/AuthStack.py

```diff
@@ -25,11 +25,14 @@ class BucketPermissions(str, Enum):
 
 
 class AuthStack(Stack):
-    def __init__(self, scope: Construct, construct_id: str, ade_iam_role: str, **kwargs) -> None:
+    def __init__(
+        self, scope: Construct, construct_id: str, ade_iam_role: str, **kwargs
+    ) -> None:
         super().__init__(scope, construct_id, **kwargs)
         self.ade_iam_role = ade_iam_role
         self.userpool = self._create_userpool()
         self.domain = self._add_domain(self.userpool)
+        self.kms_key = self._create_kms_key("stac-auth")
```

Comment:
this is the biggest change here - just create the kms key once and assign it as an attribute so it can be used by various secrets when they are created.